src/providers/fail_over.c

```diff
@@ -200,6 +200,31 @@ str_srv_data_status(enum srv_lookup_status status)
     return "unknown SRV lookup status";
 }
 
+void dump_fo_server(struct fo_server *srv)
+{
+    DEBUG(SSSDBG_OP_FAILURE, "fo_server: primary [%s] port [%d] "
+                             "port_status [%s] common->name [%s].\n",
+                             srv->primary ? "true" : "false", srv->port,
+                             str_port_status(srv->port_status),
+                             srv->common == NULL ? "common is NULL"
```

Comment:
When `srv->common` is `NULL`, the message will be `fo_server: ... common->name [common is NULL].` Wouldn't it better be just `common->name [NULL]`?

Reply:
`common->name [NULL]` would indicate that `srv->common->name` is `NULL` and `common->name [common is NULL]` means that already `srv->common` is `NULL`.

Reply:
> common->name [NULL] would indicate that srv->common->name is NULL

While most compilers will handle it well, I think strictly speaking providing NULL 'srv->common->name' to `%s` isn't "legal".

Reply:
Would it make sense to print everything from 'srv_data' and 'common' fields?